ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1103 @@
+                        rangeSelectorHight = rangeSelectorBBox.height;
+
+                    calcPosition.y += (rangeSelectorY + rangeSelectorHight) || 0;
+                }
+            }
+
+            if (positionOptions.verticalAlign === 'bottom' && chart.marginBottom) {
+                if (chart.legend && chart.legend.group && chart.legend.options.verticalAlign === 'bottom') {
+                    (calcPosition.y as number) -= chart.marginBottom -
+                            bBox.height -
+                            chart.legend.legendHeight;
+                } else {
+                    (calcPosition.y as number) -= bBox.height;
+                }
+            }
+
+            if (positionOptions.align === 'right') {
+                (calcPosition.x as number) -= bBox.width;
+            } else if (positionOptions.align === 'center') {
+                (calcPosition.x as number) -= bBox.width / 2;
+            }
+            this.breadcrumbsGroup
+                .align(
+                    calcPosition,
+                    true,
+                    chart.spacingBox
+                );
+        }
+    }
+
+    /**
+     * Calcule level on which chart currently is.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#calculateLevel
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     */
+    public calculateLevel(this: Breadcrumbs): void {
+        const chart = this.chart,
+            drilldownLevels = chart.drilldownLevels;
+
+        // Calculate on which level we are now.
+        if (this.isTreemap && (chart.series[0] as Highcharts.TreeSeries).tree) {
+            this.level = Math.abs(((chart.series[0] as Highcharts.TreeSeries).tree as any).levelDynamic) - 1;
+        } else {
+            this.level = drilldownLevels && drilldownLevels[drilldownLevels.length - 1] &&
+            drilldownLevels[drilldownLevels.length - 1].levelNumber || 0;
+        }
+    }
+
+    /**
+     * This method creates an array of arrays containing a level number
+     * with the corresponding series/point.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#createList
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @param {Highcharts.Breadcrumbs} e
+     *        Event in case of the treemap series.
+     */
+    public createList(this: Breadcrumbs, e?: any): void {
+        const breadcrumbsList: Array<Array<number|null|Point|SeriesType|SVGAElement>> = this.breadcrumbsList || [],
+            chart = this.chart,
+            drilldownLevels = chart.drilldownLevels;
+            // If the list doesn't exist treat the initial series
+            // as the current level- first iteration.
+        let currentLevelNumber: number|null = breadcrumbsList.length ?
+            (breadcrumbsList[breadcrumbsList.length - 1][0] as number) : null;
+
+        if (this.isTreemap && e) {
```

Comment:
In principle, I am not sure the Breadcrumbs class should care whether it is implemented in a Drilldown or a Treemap. The contents of `createList` and `calculateLevel` I believe belong in the Drilldown and Treemap implementations. Probably those classes should have some function that prepares the list and sends it over to the breadcrumbs, either as options, or as events if it needs to be kept updated. 

I think the litmus test for a good Breadcrumbs API would be if we can add it to a Chart and attach it to some random data structure. Can we set it up to highlight elements of a nested HTML list for example? Then we have a good modular design. If we manage that, we can drop the `isTreemap` property.

Ideally we should make it able to run without the Chart too, only depending on the Renderer, but that is probably not worthwhile.

What do you think?

Reply:
Status November 16th: Still many references to treemap and drilldown inside the Breadcrumbs, which I think should be refactored.

Reply:
I believe it should be ready now.